apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditLimitPrice.tsx

```diff
@@ -39,43 +40,34 @@ export const EditLimitPrice = () => {
         ? createOrder.selectedPool.price
         : 0
 
+    const formattedValue = BigNumber(defaultPrice).toFormat(numberLocale)
+    setText(formattedValue)
     limitPriceChanged(`${defaultPrice}`)
-
-    const sellQuantityDenominated = Quantities.denominated(
-      createOrder.amounts.sell.quantity,
-      sellTokenInfo.decimals ?? 0,
-    )
-    buyAmountChanged({
-      quantity: getBuyQuantityForLimitOrder(
-        sellQuantityDenominated,
-        asQuantity(defaultPrice ?? 0),
-        buyTokenInfo.decimals ?? 0,
-      ),
-      tokenId: createOrder.amounts.buy.tokenId,
-    })
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [isBuyTouched, isSellTouched, sellTokenInfo.id, buyTokenInfo.id, createOrder?.selectedPool?.price, numberLocale])
+  }, [
+    isBuyTouched,
+    isSellTouched,
+    sellTokenInfo.id,
+    buyTokenInfo.id,
+    createOrder.selectedPool?.price,
+    numberLocale,
+    createOrder.type,
+    limitPriceChanged,
+  ])
 
   const onChange = (text: string) => {
-    const [_, quantity] = Quantities.parseFromText(text, PRECISION, numberLocale)
+    const [newText, quantity] = Quantities.parseFromText(text, PRECISION, numberLocale)
```

Comment:
it looks like, but dunno
```suggestion
    const [formattedPrice, price] = Quantities.parseFromText(text, PRECISION, numberLocale)
```